### PR TITLE
docs (DESIGN-PATTERNS): update TS code to current version

### PR DESCRIPTION
**Pull request by dmorosinotto**
Update TS sample code to be aligned with current ngMetadata version, expeccially provide method to register directive, service, etc... in angular

### CODE REVIEW

**Comment by dmorosinotto** [comment]
HI @Hotell I'm not sure if the provider sample is correct I didn't try it... maybe there is a way to register the provider as an OpaqueToken or using the DroidSvc type ref, but I was not sure, so I stayed to the simple string 'droidSvc' that is more similar to the original js code.

PS: another observation, in [Line 92](https://github.com/ngParty/ng-metadata/blob/master/docs/DESIGN-PATTERNS.md#L92) where you stated that Directive has **no scope** I don't know if this is totally correct, because you can declare and use Inputs, Outputs and Attrs on a directive and they become "local scope" for the directive, or I'm misunderstood on it?

File: docs/DESIGN-PATTERNS.md [hunk 1]
```diff
@@ -151,7 +151,7 @@ export class ClickMe implements OnDestroy{
 
   }
 
-  onDestroy(){
+  ngOnDestroy(){
```
Comment (Hotell) [inline docs/DESIGN-PATTERNS.md]: we can go even further here. Pls get rid completely off `$element` and `OnDestroy` and just use `HostListener('click')`

**Comment by Hotell** [comment]
Good catch with the provider registration. Yeah it's kind of nasty. We can do opaque token or leave it as you updated. Anyway we shouldn't be using providers because there is no such a thing in ng2, but the design pattern may be useful for somebody

**Comment by Hotell** [comment]
once again, great job @dmorosinotto !

**Comment by dmorosinotto** [comment]
Hi @Hotell I've updated the clicker directive as you've request using the HostListener and I added a simple input just to have a sample that demonstrate something using and input attribute.

Hope you like it 😉

**Comment by dmorosinotto** [comment]
Regarding provider I know that they are not portable to NG2, but I think that it's useful to show how to port old angular code to TS and how to trick/use ngMetadata even with this nasty situation becouse anybody can learn to use TS and ngMetadata even before the NG2 migration, and so it's useful to find a way to migrate existing and valid NG1 cases.

**Comment by Hotell** [comment]
yessss, excellent! #highFive

**Comment by dmorosinotto** [comment]
:+1:  Thanks, glad you like it!

**Pull request merged**